### ISSUE TITLE
Add a badge with Fuzz Introspector metrics

### ISSUE THREAD

**Issue by ligurio** [google/oss-fuzz#10924]
There is an OSS Fuzz badge [^1], that 
> allows you to see bugs found by your OSS-Fuzz integration at a glance.

I suppose it would be useful for projects to provide badges with Fuzz Introspector metrics, say a current code coverage or fuzzers count etc.

[^1]: https://google.github.io/oss-fuzz/getting-started/new-project-guide/#status-badge

**Comment by DavidKorczynski** [comment]
I think this is a neat idea. I think in that context it may make sense to boil down the findings of Fuzz Introspector to a given score [0:100] which could be attached to the badge.

**Comment by ligurio** [comment]
> I think in that context it may make sense to boil down the findings of Fuzz Introspector to a given score [0:100] which could be attached to the badge.

I like idea!

**Comment by evverx** [comment]
I wonder if OSS-Fuzz badges could be updated instead of creating a separate badge? The FI data could be pulled if an additional parameter was passed (mostly to avoid starting showing things like `oss-fuzz: fuzzing 0.001% with 1 fuzz target` by default).

Some projects can't be linked statically for various reasons. It would be great if it was possible to show the global coverage (and other things that can be counted without FI) to cover those projects too.

**Comment by DavidKorczynski** [comment]
One option is, for example, to have a badge showing the same coverage number as on this page: https://introspector.oss-fuzz.com/projects-overview -- this number is not pulled from Fuzz Introspector but rather from  general coverage data. Even so, I'm considering whether coverage data should necessarily be shown or if it should only show (1) is the project integrated in OSS-Fuzz and (2) is the build succeeding.

**Comment by evverx** [comment]
> I'm considering whether coverage data should necessarily be shown or if it should only show (1) is the project integrated in OSS-Fuzz and (2) is the build succeeding

It's a good point. I think in general it would be great if projects could decide what they show. If they don't want to show the coverage for example (regardless of whether it builds fine or not) they could pass `?coverage=no` or something like that. As far as I can remember LGTM provided badges like that so that projects with pretty bad scores (because they just started using it) were able to show those badges anyway.

**Comment by evverx** [comment]
But generally if coverage reports fail to build it could show "N/A". I think coveralls shows either something like [![Coverage Status](https://coveralls.io/repos/github/systemd/systemd/badge.svg?branch=main)](https://coveralls.io/github/systemd/systemd?branch=main) or just "coverage: N/A".

**Comment by DavidKorczynski** [comment]
> I'm considering whether coverage data should necessarily be shown or if it should only show (1) is the project integrated in OSS-Fuzz and (2) is the build succeeding.

I just realised this is already there: https://google.github.io/oss-fuzz/getting-started/new-project-guide/#status-badge -- and @ligurio even commented this in the first post. Sorry for the confusion here, I lost track.

I will look into creating more refined badges and then users can select. Will look at both a code coverage and a fuzz introspector badge.

**Comment by evverx** [comment]
> I just realised this is already there

I think it only shows something like "coverage: failing" if OSS-Fuzz fails to build coverage reports. Generally it only shows whether builds are fine and the idea I was proposing was that it could go a little bit further and start showing how much code is covered at runtime instead of just "oss-fuzz: fuzzing". This could be used by projects failing to build with FI too.

On second thought after all maybe it would make sense to create a separate FI badge because I'm not sure how to stuff all that in oss-fuzz badges. It could be I misuse badges though :-)

**Comment by DavidKorczynski** [comment]
> start showing how much code is covered at runtime instead of just "oss-fuzz: fuzzing". This could be used by projects failing to build with FI too.
> 
> On second thought after all maybe it would make sense to create a separate FI badge because I'm not sure how to stuff all that in oss-fuzz badges. It could be I misuse badges though :-)

Yeah, I think some more detailed badges are potentially useful to have. Both code coverage (non-FI) and FI. I recon badges may be used quite differently from project to project and thus we may want to just offer a selection of them -- to accommodate the various forms of use.

**Comment by szhorvat** [comment]
> One option is, for example, to have a badge showing the same coverage number as on this page: https://introspector.oss-fuzz.com/projects-overview -- this number is not pulled from Fuzz Introspector but rather from general coverage data. Even so, I'm considering whether coverage data should necessarily be shown or if it should only show (1) is the project integrated in OSS-Fuzz and (2) is the build succeeding.

IMO coverage numbers are not comparable between projects. Currently, igraph has a number of 17% on this page, but I would say it's coverage is actually excellent, as good as it can possibly get.  This is because not all parts of the library are suitable for fuzzing. We fuzz the parsers for various graph storage formats, and put a lot of effort into making sure that they are fuzzed thoroughly. But at this point I am becoming more and more convinced that trying to fuzz graph algorithms, which make up the majority of the library, is not possible / not productive. See also the discussion I started here: https://github.com/google/oss-fuzz/discussions/11088

**Comment by evverx** [comment]
> This is because not all parts of the library are suitable for fuzzing

I think there is an FI issue somewhere where attack surfaces were discussed. The idea was to make it possible to weed out parts that aren't important in terms of fuzzing as far as I can remember and focus on things that should be fuzzed and how good they are (as opposed to the full coverage).
